src/libraries/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/Windows/PkcsPalWindows.cs

```diff
@@ -93,27 +93,27 @@ public sealed override byte[] GetSubjectKeyIdentifier(X509Certificate2 certifica
 
         public override T GetPrivateKeyForSigning<T>(X509Certificate2 certificate, bool silent)
         {
-            return GetPrivateKey<T>(certificate, silent, preferNCrypt: true);
+            return GetPrivateKey<T>(certificate, silent, preferNCrypt: true)!; // It seems we can add 'where T : AsymmetricAlgorithm' and return nullable
```

Comment:
I don't think the comment has long-term value.

Doesn't `[MaybeNull]` solve this problem?  (I'm not sure why the generic constraint can't be applied; but if it can't, [MaybeNull] seems like it solves the return type nullability).

Reply:
That comment is my question, not intended to keep, yes i can use [MaybeNull] just thought its better to make it restricted to a class and make nullable